dist/tools/testrunner/testrunner.py

```diff
@@ -38,9 +39,15 @@ def run(testfunc, timeout=10, echo=True, traceback=False):
         if traceback:
             print_tb(sys.exc_info()[2])
         return 1
+    except EOF:
+        print("EOF in expect script")
```

Comment:
Maybe "Unexpected end of file in expect script"? Maybe you can also output the expect it is failing on, like I did in #7845.